### PR TITLE
Add console logging option

### PR DESCRIPTION
**Pull request by paramite**

### CODE REVIEW

**Comment by paramite** [comment]
@pleimer Seems like you accidentally submitted PR to my personal repo. Created the PR here instead. Please resolve conflicts, so we can merge it.

**Comment by leifmadsen** [comment]
I guessed and resolved the conflict :)

File: logging/logger.go [hunk 1]
```diff
@@ -16,62 +18,72 @@ const (
...
 	logger.Timestamp = false
 	logger.metadata = make(map[string]interface{})
 
-	logfile, err := os.OpenFile(path, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
-	if err != nil {
-		return nil, err
+	switch strings.ToLower(target) {
+	case "console":
+		logger.write = func(message string) error {
+			fmt.Print(message)
```
Comment (paramite) [inline logging/logger.go:46]: Println maybe? Input logs don't contain new line IIRC, so this will make the whole program run output single line of logs.

**Review by paramite — COMMENTED** [review]
Except the writeline issue I am fine with this patch. Question is if this is not a bit over-engineering though. Pointing to stdout or stderr (as we do within Lokean [1]) does the same job. But anyway, I don't mind merging this once we don't write all logs to single line.

[1] https://github.com/paramite/collectd-sensubility/pull/4

File: logging/logger.go [hunk 2]
```diff
@@ -16,62 +18,72 @@ const (
...
-
-func (self *Logger) Metadata(metadata map[string]interface{}) {
-	self.metadata = metadata
-}
-
-func(self *Logger) SetLogLevel(level LogLevel) {
-	self.Level = level
-}
-
-func(self *Logger) SetFile(path string, permissions os.FileMode) error {
```
Comment (paramite) [inline logging/logger.go:57]: Oh yes, almost missed that. Please don't remove this method as it is used in at least one project already.

Comment (pleimer) [reply logging/logger.go:57]: nice catch. May have accidentaly deleted it

File: logging/logger.go (hunk 1 above)
Comment (pleimer) [reply logging/logger.go:46]: writeRecord() adds the newline here: https://github.com/infrawatch/apputils/blob/f13cb337a60cd8f74d0747a333f993716a3fc705/logging/logger.go#L128

**Comment by pleimer** [comment]
> Except the writeline issue I am fine with this patch. Question is if this is not a bit over-engineering though. Pointing to stdout or stderr (as we do within Lokean [1]) does the same job. But anyway, I don't mind merging this once we don't write all logs to single line.
> 
> [1] https://github.com/infrawatch/lokean/blob/master/cmd/main.go#L39

That's fair I hadn't thought of doing that when I originally made this patch to use in ci-server-go. If you want to stick with that, that works too.

**Comment by pleimer** [comment]
I guess the only benefit to doing it this way is that it works on windowsl, though it may be as easy as pointing to some file stream there as well

File: logging/logger.go (hunk 1 above)
Comment (paramite) [reply logging/logger.go:46]: Ah, right. Then just ignore my comment :).

**Pull request merged**